### PR TITLE
Implement symmetric difference on primitive sets. Partially addresses #310.

### PR DESCRIPTION
**Pull request by prathasirisha**
Signed-off-by: Sirisha Pratha <sirisha.pratha@bnymellon.com>

### CODE REVIEW

**Review by donraab — APPROVED** [review]
LGTM. Thanks for the contribution @prathasirisha!

File: eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg [hunk 1]
```diff
@@ -90,6 +90,19 @@ public interface Immutable<name>Set extends Immutable<name>Collection, <name>Set
...
+     * Returns the set of all objects that are a member of exactly one of {@code this} and {@code set} (elements which
+     * are in one of the sets, but not in both). For instance, for the sets [1, 2, 3] and [2, 3, 4], the symmetric
+     * difference set is [1, 4] . It is the set difference of the union and the intersection.
+     *
+     * @since 11.0.
+     */
+    @Override
+    default Immutable<name>Set symmetricDifference(<name>Set set)
+    {
+        return this.toSet().symmetricDifference(set).toImmutable();
```
Comment (donraab) [inline eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg:103]: Hi @prathasirisha, we might be able to optimize this slightly.

`return set.reject(this::contains, this.reject(set::contains, Sets.mutable.empty())).toImmutable();`

Comment (mohrezaei) [reply eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg:103]: `Sets` is not for primitives, no?

Comment (donraab) [reply eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg:103]: Oh right... and we don't have this on primitive side yet.

Comment (donraab) [reply eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg:103]: @mohrezaei, any other suggestions on the code for potential optimizations?

Comment (mohrezaei) [reply eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg:103]: @donraab nope. LGTM

**Pull request merged**